contracts/Presale.sol

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity =0.8.9;
+
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+
+import "./IPresale.sol";
+
+contract Presale is IPresale, Ownable2Step, ReentrancyGuard, Pausable {
+    address public immutable USDC;
+
+    address public immutable DAI;
+
+    AggregatorV3Interface public immutable ORACLE;
+
+    uint256 private immutable PRECISION = 1e18;
+    uint256 private immutable ETH_TO_WEI_PRECISION = 1e10;
+    uint256 private immutable USDC_TO_WEI_PRECISION = 1e12;
+
+    uint256 private $currentRoundIndex;
+    uint256 private $totalRaisedUSD;
+
+    PresaleConfig private $config;
+    RoundConfig[] private $rounds;
+
+    mapping(uint256 => uint256) private $raisedUSD;
```

Comment:
Because we are dealing with tokens. It would be easier to have ths one as the number of tokens that the round has currently exhuasted
```suggestion
    mapping(uint256 => uint256) private $roundAllocated;
```